front/src/core-logic/domain/admin/icUsersAdmin/icUsersAdmin.epics.ts

```diff
@@ -121,6 +121,58 @@ const rejectAgencyToUserEpic: IcUsersAdminActionEpic = (
     ),
   );
 
+const createUserOnAgencyEpic: IcUsersAdminActionEpic = (
+  action$,
+  state$,
+  { adminGateway },
+) =>
+  action$.pipe(
+    filter(icUsersAdminSlice.actions.createUserOnAgencyRequested.match),
+    switchMap((action) =>
+      adminGateway
+        .createUserForAgency$(action.payload, getAdminToken(state$.value))
+        .pipe(
+          map(
+            ({
+              id,
+              email,
+              firstName,
+              lastName,
+              createdAt,
+              dashboards,
+              externalId,
+              agencyRights,
+            }) =>
+              icUsersAdminSlice.actions.createUserOnAgencySucceeded({
+                icUser: {
+                  id,
+                  email,
+                  firstName,
+                  lastName,
+                  externalId,
+                  createdAt,
+                  dashboards,
+                  agencyRights: agencyRights.reduce(
+                    (agenciesAcc, agencyRight) => ({
+                      ...agenciesAcc,
+                      [agencyRight.agency.id]: agencyRight,
+                    }),
+                    {} as Record<AgencyId, AgencyRight>,
+                  ),
+                },
+                feedbackTopic: action.payload.feedbackTopic,
+              }),
```

Comment:
on peut peut-être alleger un peu l'écriture :

```suggestion
            (user) =>
              icUsersAdminSlice.actions.createUserOnAgencySucceeded({
                icUser: {
                  ...user,
                  agencyRights: user.agencyRights.reduce(
                    (agenciesAcc, agencyRight) => ({
                      ...agenciesAcc,
                      [agencyRight.agency.id]: agencyRight,
                    }),
                    {} as Record<AgencyId, AgencyRight>,
                  ),
                },
                feedbackTopic: action.payload.feedbackTopic,
              }),
```